keep-alive/src/test/java/KeepAliveGattTest.kt

```diff
@@ -350,6 +354,148 @@ class KeepAliveGattTest {
         }
     }
 
+    @Test
+    fun `When 'connect' occurs, the onConnect lambda is called once`() = runBlocking {
+        val bluetoothDevice = mockBluetoothDevice()
+        val gatt1 = mockk<Gatt> {
+            every { onCharacteristicChanged } returns flow { delay(Long.MAX_VALUE) }
+            coEvery { disconnect() } returns Unit
+        }
+
+        mockkStatic(BLUETOOTH_DEVICE_CLASS) {
+            var onConnectLambdaCalls = 0
+            coEvery {
+                bluetoothDevice.connectGatt(any())
+            } returns ConnectGattResult.Success(gatt1)
+
+            val job = Job()
+            val keepAlive = CoroutineScope(job).keepAliveGatt(
+                androidContext = mockk(relaxed = true),
+                bluetoothDevice = bluetoothDevice,
+                disconnectTimeoutMillis = DISCONNECT_TIMEOUT,
+                onEventAction = {
+                    onConnected {
```

Comment:
Looking at the API in use here. I think it'll actually read a little better as:

```suggestion
                onEvent = { event ->
                    event.onConnected {
```

So we'd change the `OnEventAction` naming to be just `OnEvent`. Since `suspend` implies that developers can suspend if they want in those lambdas? So the "action" becomes superfluous.

Rather than change the `OnEvent` scope to be `Event`, we have the `event` be a parameter. I think it makes the lambda a little more verbose, but more clearer.

Thoughts?

Reply:
You're right - I was thinking the 'Action' terminology was a little awkward here but I couldn't think of a better approach.

I might go ahead and replace the typealias for EventAction as well because it's only used in a couple spots and I think it reads more clearly without it in place.